rest_spec.md

```diff
@@ -63,15 +63,15 @@ These APIs allow the admin user to create new users with access to certain resou
 
 | HTTP Verb | URL | Request Body | Response Code | Response Body | Function |
 | --- | --- | :---  | :---  | :---  | :--- |
-| POST | `/users` | `{"roleSetId": "client\|admin","resourceId": "*\|some id"}` | See Collection responses | `{“bearerToken”: "token"}` | Create a new user and assign a bearer token, setup internal management of the user-id that does not require saving the bearer token and attached the named `roleSet` for the `resource-id` to the new user |
-| DELETE | `/users/user-id` | `{"roleSetId": "client\|admin", "resourceId": "*\|<some-user-id>"}` | See Item responses | `{"bearerToken": "token"}` | Removes the user and all access rules from the system |
+| POST | `/users/user-id/permissions` | `{"userId": "user-id", "roleSetId": "client\|admin","resourceId": "*\|some id"}` | See Collection responses | `{"userId": "user_id", “bearerToken”: "token"}` | Create a new user and assign a bearer token and user-id, setup internal management of the user-id that does not require saving the bearer token and attached the named `roleSet` for the `resource-id` to the new user |
```

Comment:
If we are creating a user-id then this should be `POST /users` since the user-id is not known until it is created or we need both `/users` to return the user-id and `/users/user-id/permissions` to create permissions for the user-id. 

I assume permissions is a collection of roleSets keyed to resourceId? So a user-id can write to several engine-ids by setting several permissions on the same user-id?

How are we creating the user-id/client-id? It should be the server creating the id, right?